File: typescript/typelayer-c/bin/schemas/world/structures.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Module: schemas.world.structures
const typelayerCommonDocumentation = __importStar(require("typelayer-common"));
class File {
    constructor(arg) {
        Object.assign(this, arg);
    }
}
exports.File = File;
function defaultFile() {
    let contentsVal;
    let fileidVal;
    contentsVal = "";
    fileidVal = "";
    return new File({ contents: contentsVal, fileid: fileidVal });
}
exports.defaultFile = defaultFile;
function describeFile() {
    let contentsFieldInfo;
    let fileConstructorInfo;
    let fileContentsFieldInfo;
    let fileFileidFieldInfo;
    let fileidFieldInfo;
    contentsFieldInfo = new typelayerCommonDocumentation.TlText({});
    fileContentsFieldInfo = new typelayerCommonDocumentation.FieldDescription({
        name: "contents",
        description: null,
        typeInfo: contentsFieldInfo
    });
    fileidFieldInfo = new typelayerCommonDocumentation.TlText({});
    fileFileidFieldInfo = new typelayerCommonDocumentation.FieldDescription({
        name: "fileid",
        description: null,
        typeInfo: fileidFieldInfo
    });
    fileConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({
        name: "file",
        description: null,
        fields: [fileContentsFieldInfo, fileFileidFieldInfo]
    });
    return new typelayerCommonDocumentation.StructureInfo({
        structureId: "file",
        structureType: new typelayerCommonDocumentation.StructureEnum({}),
        description: null,
        constructors: [fileConstructorInfo]
    });
}
exports.describeFile = describeFile;

File: haskell/typelayer-c/src/TypelayerC/Schemas/World/Structures.hs
{-# LANGUAGE GADTs, OverloadedStrings, StandaloneDeriving, DeriveAnyClass, GeneralisedNewtypeDeriving #-}
module TypelayerC.Schemas.World.Structures where
import qualified TypelayerCommon.Documentation as TypelayerCommonDocumentation
import qualified TypelayerCommon.Prelude as TypelayerCommonPrelude
import qualified TypelayerCommon.Types as TypelayerCommonTypes


data File = 
    File
        { fileContents :: TypelayerCommonTypes.Text
        , fileFileid :: TypelayerCommonTypes.Text
        }
    deriving (Show, Eq)

defaultFile :: TypelayerCommonPrelude.ExecuteM File
defaultFile  = do
    let contentsVal = ""
    let fileidVal = ""
    return (File{fileContents=contentsVal, fileFileid=fileidVal})

describeFile :: TypelayerCommonPrelude.ExecuteM TypelayerCommonDocumentation.StructureInfo
describeFile  = do
    let contentsFieldInfo = (TypelayerCommonDocumentation.TlText{})
    let fileContentsFieldInfo = (TypelayerCommonDocumentation.FieldDescription{TypelayerCommonDocumentation.fieldDescriptionName="contents", TypelayerCommonDocumentation.fieldDescriptionDescription=Nothing, TypelayerCommonDocumentation.fieldDescriptionTypeInfo=contentsFieldInfo})
    let fileidFieldInfo = (TypelayerCommonDocumentation.TlText{})
    let fileFileidFieldInfo = (TypelayerCommonDocumentation.FieldDescription{TypelayerCommonDocumentation.fieldDescriptionName="fileid", TypelayerCommonDocumentation.fieldDescriptionDescription=Nothing, TypelayerCommonDocumentation.fieldDescriptionTypeInfo=fileidFieldInfo})
    let fileConstructorInfo = (TypelayerCommonDocumentation.ConstructorInfo{TypelayerCommonDocumentation.constructorInfoName="file", TypelayerCommonDocumentation.constructorInfoDescription=Nothing, TypelayerCommonDocumentation.constructorInfoFields=[fileContentsFieldInfo, fileFileidFieldInfo]})
    return (TypelayerCommonDocumentation.StructureInfo{TypelayerCommonDocumentation.structureInfoStructureId="file", TypelayerCommonDocumentation.structureInfoStructureType=(TypelayerCommonDocumentation.StructureEnum{}), TypelayerCommonDocumentation.structureInfoDescription=Nothing, TypelayerCommonDocumentation.structureInfoConstructors=[fileConstructorInfo]})